### DISCUSSION TITLE
influxdb: column order of reader

### DISCUSSION THREAD

**Discussion by IronsDu** [wgzhao/Addax#163] [General]
**Describe the bug**
we can't assume the column's order of response by influxdb.  so we need specify the order in config.
like this:
```sh
"reader": {
                    "name": "influxdbreader",
                    "parameter": {
                        "column": [
                            "user_name",
                            "time",
                            "user_id",
                        ],
                        "connection": [
                            {
                                "endpoint": "http://localhost:8086",
                                "database": "datax",
                                "table": "datax_tb2",
                            }
                        ],
                        "username": "jdoe",
                        "password": "abcABC123!@#",
                        "querySql": "select user_name, user_id, xxxxxxxx from  datax_tb2",
                    }
                },
```
use this config, we output record data by this column order : [user_name, time, user_id]
**To Reproduce**
1. paste  json content of running job
2. paste the output of running job

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. CentOS]
 - JDK Version [e.g. 1.8]
 - DataX Version [e.g. 3.1.3]

**Additional context**

**Comment by IronsDu** [comment]
也就是说我们期望能定义column的顺序。那么select之后拿到response时，我们能按照这个顺序去组织输出的record。
@wgzhao 你觉得这是一个问题么？

**Comment by wgzhao** [comment]
@IronsDu 你在实际使用过程中出现过字段顺序和表定义顺序不一致的情况吗？

**Comment by IronsDu** [comment]
因为我们无法确定influxdb的返回顺序。 所以假设当我们从influxdb迁移到MySQL，那我们就得猜测(或通过测试来得到)这个顺序。

**Comment by wgzhao** [comment]
返回的字段是确保有序的，如果是 "*"，则按照表的定义顺序，否则按照查询顺序，这一点是和其他RDBMS执行SQL查询是相似的。拿到的结果是一个list，组装record时，也是按照list的顺序来组织的。

**Comment by IronsDu** [comment]
我说的是从influxdb里query出来的顺序。

**Comment by wgzhao** [comment]
> 我说的是从influxdb里query出来的顺序，

这个是有序的

**Comment by IronsDu** [comment]
嗯，但那是字典序。我在想是否能直接在confi里配置输入和输出的字段映射顺序（关系）？ （这样更具有确定性和方便性，也减少influxdb的实现依赖）

**Reply by wgzhao** [reply]
你这里的输入是指查询时的字段，输出时response的字段吗？
映射关系是指？

还是说比如你是要influxdb到mysql，其中influxdb的顺序是`col1,col2,col3`, 而mysql表的字段是 `col3,col1,col2`

能否给出一个具体的场景来描述这个问题？

**Comment by IronsDu** [comment]
请问现在influxdb reader 的conf里配置COLUMN的作用是什么呢？
